clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -104,18 +104,19 @@
     public static final String MIN_COMPACTION_LAG_MS_DOC = "The minimum time a message will remain " +
         "uncompacted in the log. Only applicable for logs that are being compacted.";
 
+    public static final String MAX_COMPACTION_LAG_MS_CONFIG = "max.compaction.lag.ms";
+    public static final String MAX_COMPACTION_LAG_MS_DOC = "The maximum time a message will remain " +
+        "ineligible for compaction in the log. Only applicable for logs that are being compacted.";
+
     public static final String MIN_CLEANABLE_DIRTY_RATIO_CONFIG = "min.cleanable.dirty.ratio";
     public static final String MIN_CLEANABLE_DIRTY_RATIO_DOC = "This configuration controls how frequently " +
         "the log compactor will attempt to clean the log (assuming <a href=\"#compaction\">log " +
         "compaction</a> is enabled). By default we will avoid cleaning a log where more than " +
         "50% of the log has been compacted. This ratio bounds the maximum space wasted in " +
         "the log by duplicates (at 50% at most 50% of the log could be duplicates). A " +
         "higher ratio will mean fewer, more efficient cleanings but will mean more wasted " +
-        "space in the log.";
-
-    public static final String MAX_COMPACTION_LAG_MS_CONFIG = "max.compaction.lag.ms";
-    public static final String MAX_COMPACTION_LAG_MS_DOC = "The maximum time a message will remain " +
-        "ineligible for compaction in the log. Only applicable for logs that are being compacted.";
+        "space in the log. If " + MAX_COMPACTION_LAG_MS_CONFIG + " is used, the log compactor will attempt to clean " +
```

Comment:
`attempt to clean` -> `consider the log to be eligible for compaction`. That said, I don't think you need to mention this here. If you do, you may want to cross-mention this across all the related threshold configurations - i.e., min.compaction.lag (and mention that it has to be less than max.compaction.lag); here; and then in max.compaction.lag.ms.doc mention its relationship to min.cleanable and min.compaction.lag

So the simpler alternative, is to just stick to the existing convention. Document each of these individually, and summarize the net behavior in the main design html.